modules/cas_disk/exp_obj.c

```diff
@@ -505,61 +535,54 @@ int casdsk_exp_obj_create(struct casdsk_disk *dsk, const char *dev_name,
 
 	result = casdsk_exp_obj_alloc(dsk);
 	if (result)
-		goto error_alloc;
+		goto error_exp_obj_alloc;
 
 	exp_obj = dsk->exp_obj;
 
 	exp_obj->dev_name = kstrdup(dev_name, GFP_KERNEL);
 	if (!exp_obj->dev_name) {
 		__casdsk_exp_obj_release(exp_obj);
 		result = -ENOMEM;
-		goto error_strdup;
+		goto error_kstrdup;
```

Comment:
jumping to error_kstrdup will call casdsk_exp_obj_free and put the kobject ("kobject_put(&exp_obj->kobj)") while the kobject is initialized later in _casdsk_exp_obj_init_kobject().

Reply:
agreed, reworked